### ISSUE TITLE
[Bug]: Client list_collections gets empty result even though the milvus does have a collection

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#36889]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.4-20241013-44564f04-amd64
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. milvus cluster is healthy (deployed by milvus-operator)
```
compact-opt-rate-100m-1-milvus-datanode-5895486c79-kswl8          1/1     Running     0               68m     10.104.1.232    4am-node10   <none>           <none>
compact-opt-rate-100m-1-milvus-indexnode-559f74bbfd-dqfcg         1/1     Running     0               65m     10.104.34.117   4am-node37   <none>           <none>
compact-opt-rate-100m-1-milvus-indexnode-559f74bbfd-td8k2         1/1     Running     0               65m     10.104.19.208   4am-node28   <none>           <none>
compact-opt-rate-100m-1-milvus-indexnode-559f74bbfd-tn72z         1/1     Running     0               68m     10.104.33.11    4am-node36   <none>           <none>
compact-opt-rate-100m-1-milvus-mixcoord-86858db78b-thd5z          1/1     Running     0               67m     10.104.18.149   4am-node25   <none>           <none>
compact-opt-rate-100m-1-milvus-proxy-5b9ccc5778-7q9bp             1/1     Running     0               67m     10.104.23.224   4am-node27   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-dpv4q       1/1     Running     0               65m     10.104.4.241    4am-node11   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-kvcmc       1/1     Running     0               65m     10.104.15.60    4am-node20   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-p2bz9       1/1     Running     0               65m     10.104.25.81    4am-node30   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-zzkpr       1/1     Running     0               69m     10.104.21.72    4am-node24   <none>           <none>
```
2. cluster has a collection `fouram_rf7If1S0` and loaded edit mi and upgrade queryNode.replicas 4 -> 5, add node-selector
```
    queryNode:
      nodeSelector:
        node-role/nvme: "true"
      replicas: 5
```
3. wait healthy
```
zong@nausicca:~$ k delete pod -n qa-milvus compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-kvcmc
pod "compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-kvcmc" deleted
^C
zong@nausicca:~$ k delete pod -n qa-milvus compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-kvcmc --force
warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.
pod "compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-kvcmc" force deleted
zong@nausicca:~$ k delete pod -n qa-milvus compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-dpv4q --force
warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.
pod "compact-opt-rate-100m-1-milvus-querynode-1-7f669bc6fc-dpv4q" force deleted

compact-opt-rate-100m-1-milvus-datanode-84789fbbf-55kdn           1/1     Running     0               81m     10.104.13.130   4am-node16   <none>           <none>
compact-opt-rate-100m-1-milvus-indexnode-746788984-4wxw5          1/1     Running     0               80m     10.104.30.33    4am-node38   <none>           <none>
compact-opt-rate-100m-1-milvus-indexnode-746788984-6brwr          1/1     Running     0               81m     10.104.6.18     4am-node13   <none>           <none>
compact-opt-rate-100m-1-milvus-indexnode-746788984-rvk8b          1/1     Running     0               80m     10.104.1.236    4am-node10   <none>           <none>
compact-opt-rate-100m-1-milvus-mixcoord-cd9c5f87b-srssb           1/1     Running     0               81m     10.104.5.247    4am-node12   <none>           <none>
compact-opt-rate-100m-1-milvus-proxy-556fd66bdd-xpmcw             1/1     Running     0               81m     10.104.15.144   4am-node20   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-0-88b5bfd8f-6mxrg        1/1     Running     0               81m     10.104.34.149   4am-node37   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-0-88b5bfd8f-7xmxw        1/1     Running     0               49m     10.104.33.211   4am-node36   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-0-88b5bfd8f-9ctp5        1/1     Running     0               108s    10.104.15.224   4am-node20   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-0-88b5bfd8f-h9g7m        1/1     Running     0               18m     10.104.26.107   4am-node32   <none>           <none>
compact-opt-rate-100m-1-milvus-querynode-0-88b5bfd8f-sdnxt        1/1     Running     0               80m     10.104.18.121   4am-node25   <none>           <none>
```
4. milvus be healthy but list_collections() return empty []. See birdwatcher, the collection is existed
```
Milvus(compact-opt-rate-100m-1) > show collections
================================================================================
DBID: 1
Collection ID: 452746410398065847	Collection Name: fouram_rf7If1S0
Collection State: CollectionCreated	Create Time: 2024-09-23 19:52:45
Fields:
 - Field ID: 0 	 Field Name: RowID 	 Field Type: Int64
 - Field ID: 1 	 Field Name: Timestamp 	 Field Type: Int64
 - Field ID: 100 	 Field Name: id 	 Field Type: VarChar
	 - Primary Key: true, AutoID: false
	 - Type Param max_length: 128
 - Field ID: 101 	 Field Name: float_vector 	 Field Type: FloatVector
	 - Type Param dim: 128
Enable Dynamic Schema: false
Consistency Level: Bounded
Start position for channel compact-opt-rate-100m-1-rootcoord-dml_0(compact-opt-rate-100m-1-rootcoord-dml_0_452746410398065847v0): [8 235 2 16 0 24 0 32 0]
Collection properties(0):
================================================================================
--- Total collections:  1	 Matched collections:  1
--- Total channel: 1	 Healthy collections: 1
```
6. cp lag
![image](https://github.com/user-attachments/assets/5f9b3cac-d864-4373-b939-69b3ba9f0eff)

[metrics](https://grafana-4am.zilliz.cc/d/uLf5cJ3Ga/milvus2-0?orgId=1&var-datasource=db7e3557-d29c-4d2b-be40-9f465bcec112&var-namespace=qa-milvus&var-instance=compact-opt-rate-100m-1&var-collection=All&var-app_name=milvus&from=1728974396027&to=1728986635758)

7. rootCoord error logs
![image](https://github.com/user-attachments/assets/4bd26e61-3abb-432c-9f46-486d5627b762)

**Comment by congqixia** [comment]
Root cause is #36933

**Comment by congqixia** [comment]
patch merged, could you please verify?
/assign @ThreadDao

**Comment by ThreadDao** [comment]
fixed hotfix-2.4.13-20241017-dde79796-amd64
